### ISSUE TITLE
3.0 - Errors when importing orders

### ISSUE THREAD

**Issue by robincornett** [easydigitaldownloads/easy-digital-downloads#8137]
### Expected behavior
When I import a CSV of orders, I expect it to run without errors.

### Actual behavior
I get errors:

```
[30-Sep-2020 13:56:30 UTC] PHP Notice:  Undefined property: wpdb::$customermeta in \wp-includes\wp-db.php on line 648
[30-Sep-2020 13:56:30 UTC] PHP Stack trace:
[30-Sep-2020 13:56:30 UTC] PHP   1. {main}() \wp-admin\admin-ajax.php:0
[30-Sep-2020 13:56:30 UTC] PHP   2. do_action() \wp-admin\admin-ajax.php:176
[30-Sep-2020 13:56:30 UTC] PHP   3. WP_Hook->do_action() \wp-includes\plugin.php:478
[30-Sep-2020 13:56:30 UTC] PHP   4. WP_Hook->apply_filters() \wp-includes\class-wp-hook.php:311
[30-Sep-2020 13:56:30 UTC] PHP   5. edd_do_ajax_import() \wp-includes\class-wp-hook.php:287
[30-Sep-2020 13:56:30 UTC] PHP   6. EDD_Batch_Payments_Import->process_step() \wp-content\plugins\easy-digital-downloads.30\includes\admin\import\import-functions.php:140
[30-Sep-2020 13:56:30 UTC] PHP   7. EDD_Batch_Payments_Import->create_payment() \wp-content\plugins\easy-digital-downloads.30\includes\admin\import\class-batch-import-payments.php:112
[30-Sep-2020 13:56:30 UTC] PHP   8. EDD_Batch_Payments_Import->set_customer() \wp-content\plugins\easy-digital-downloads.30\includes\admin\import\class-batch-import-payments.php:169
[30-Sep-2020 13:56:30 UTC] PHP   9. wpdb->__get() \wp-content\plugins\easy-digital-downloads.30\includes\admin\import\class-batch-import-payments.php:403
[30-Sep-2020 13:56:30 UTC] WordPress database error You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'WHERE meta_key = '_canonical_import_id' AND meta_value = 1 LIMIT 1' at line 1 for query SELECT customer_id FROM  WHERE meta_key = '_canonical_import_id' AND meta_value = 1 LIMIT 1 made by do_action('wp_ajax_edd_do_ajax_import'), WP_Hook->do_action, WP_Hook->apply_filters, edd_do_ajax_import, EDD_Batch_Payments_Import->process_step, EDD_Batch_Payments_Import->create_payment, EDD_Batch_Payments_Import->set_customer
```

This is a 3.0 > 3.0 export/import; the behavior may be different exporting from 2.9 and importing to 3.0.

### Steps to reproduce the behavior
1) In 3.0, visit Downloads > Reports > Export.
2) Use "Export Orders" to export all store orders as a CSV.
3) Visit Downloads > Tools.
4) Reset the store (optional).
5) Visit Downloads > Tools > Import/Export.
6) Import the CSV using "Import Payment History" _(question: should this be "Import Orders" instead?)_

### Information (if a specific version is affected):
PHP Version: 7.4.1
EDD Version (or branch): release/3.0
WordPress Version: 5.5.1

Any other relevant information:
The orders are importing, but the customer information is wonky--names are missing, and the "customer since" date is the date of the import, not the date of the first order. So all orders are showing "No Name" for the customer in the orders table, although the customer is selected in the order edit screen. This should also be done with full order information (my store is orders generated via CLI, so there is no address information).

**Comment by ashleyfae** [comment]
In EDD master's `EDD_DB_Customer_Meta` class:

```php
public function register_table() {
	global $wpdb;
	$wpdb->customermeta = $this->table_name;
}
```

Off the top of my head, I think BerlinDB makes this `$wpdb->edd_customermeta`, but we will want to keep the old one for backwards compatibility.

---

> Import the CSV using "Import Payment History" (question: should this be "Import Orders" instead?)

I think so :+1:

**Comment by robincornett** [comment]
One issue is that in 2.9, the customer meta table uses `customer_id` as the column name; 3.0 uses `edd_customer_id` instead. Updating all `customer_id` in the import batch tool to `edd_customer_id` and changing one instance of `$wpdb->customermeta` to `$wpdb->edd_customermeta` resolves the errors getting logged. Wanting to confirm that the table column change is intentional and correct please @ashleyfae.

Another issue is that the export tool is not correctly exporting the first/last name for the customer if there is no address for the order, which is possible. I can fix this, too, but likely a separate issue. The export report is using the name information from the address, but we should probably use `edd_get_customer( $order->customer_id )` instead, although that returns just a name, not first and last name separately.

**Comment by robincornett** [comment]
I do see that the [column change was intentional](https://github.com/easydigitaldownloads/easy-digital-downloads/commit/6e108581fcbea2f686aeb15b6d71163c10b5b03a) but can't tell why from the commit.

**Comment by ashleyfae** [comment]
> One issue is that in 2.9, the customer meta table uses `customer_id` as the column name; 3.0 uses `edd_customer_id` instead. Updating all `customer_id` in the import batch tool to `edd_customer_id` and changing one instance of `$wpdb->customermeta` to `$wpdb->edd_customermeta` resolves the errors getting logged. Wanting to confirm that the table column change is intentional and correct please @ashleyfae.

Yes that was intentional. It's because we use the metadata API built into WordPress core. With that, you pass in a `meta_type` value, which should be unique. See the `edd_add_customer_meta()` function. Our meta_type is `edd_customer`. WordPress core then automatically builds queries for you, including putting together the "object_id" column name, which ends up being:

```php
$this->meta_id_column = sanitize_key( $type . '_id' );
```

So that turns into `edd_customer_id`.

tl;dr: It's required compatibility with the WordPress core metadata API.

**Comment by ashleyfae** [comment]
@robincornett I'm also getting this error:

```
[15-Oct-2020 12:49:02 UTC] PHP Notice:  Undefined variable: email in /var/www/edd3/wp-content/plugins/easy-digital-downloads/includes/admin/import/class-batch-import-payments.php on line 497
```

Can you fix that as well while we're here?
